kornia/color/gray.py

```diff
@@ -27,11 +29,23 @@ def rgb_to_grayscale(image: torch.Tensor) -> torch.Tensor:
         raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
                          .format(image.shape))
 
+    if not isinstance(rgb_weights, torch.Tensor):
+        raise TypeError("rgb_weights is not a torch.Tensor. Got {}".format(
+            type(rgb_weights)))
+
+    if len(rgb_weights.shape) != 1 or rgb_weights.shape[0] != 3:
+        raise ValueError("rgb_weights must have a shape of (3). Got {}"
+                         .format(rgb_weights.shape))
+
+    if not torch.isclose(torch.sum(rgb_weights), torch.tensor(1.0)):
+        raise ValueError("The sum of rgb_weights must be 1. Got {}"
+                         .format(torch.sum(rgb_weights)))
+
     r: torch.Tensor = image[..., 0:1, :, :]
     g: torch.Tensor = image[..., 1:2, :, :]
     b: torch.Tensor = image[..., 2:3, :, :]
 
-    gray: torch.Tensor = 0.299 * r + 0.587 * g + 0.114 * b
+    gray: torch.Tensor = rgb_weights[0] * r + rgb_weights[1] * g + rgb_weights[2] * b
```

Comment:
```suggestion
    gray: torch.Tensor = rgb_weights[..., 0] * r + rgb_weights[..., 1] * g + rgb_weights[..., 2] * b
```
I would suggest in that way so that the user could even provide a weight per pixel